src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -70,70 +64,23 @@ public class SamIam implements IamProviderInterface {
 
   private final SamConfiguration samConfig;
   private final ConfigurationService configurationService;
+  private final SamApiService samApiService;
 
   // This value is the same for all environments which is why this is hardcoded instead of config
   static final String TOS_URL = "app.terra.bio/#terms-of-service";
 
   @Autowired
-  public SamIam(SamConfiguration samConfig, ConfigurationService configurationService) {
+  public SamIam(
+      SamConfiguration samConfig,
+      ConfigurationService configurationService,
+      SamApiService samApiService) {
     this.samConfig = samConfig;
     this.configurationService = configurationService;
+    this.samApiService = samApiService;
   }
 
   private static final Logger logger = LoggerFactory.getLogger(SamIam.class);
 
-  private ApiClient getApiClient(String accessToken) {
-    ApiClient apiClient = new ApiClient();
-    apiClient.setAccessToken(accessToken);
-    apiClient.setUserAgent("OpenAPI-Generator/1.0.0 java"); // only logs an error in sam
-
-    // Sometimes Sam calls can take longer than the OkHttp default of 10 seconds to return a
-    // response.  In those cases, we can see socket timeout exceptions despite the underlying Sam
-    // call continuing to execute and possibly succeeding.
-    int operationTimeoutSeconds =
-        configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS);
-    apiClient.setReadTimeout((int) Duration.ofSeconds(operationTimeoutSeconds).toMillis());
-
-    return apiClient.setBasePath(samConfig.getBasePath());
-  }
-
-  private ApiClient getUnauthApiClient() {
-    ApiClient apiClient = new ApiClient();
-    apiClient.setUserAgent("OpenAPI-Generator/1.0.0 java"); // only logs an error in sam
-    apiClient.setBasePath(samConfig.getBasePath());
-    return apiClient;
-  }
-
-  @VisibleForTesting
-  public ResourcesApi samResourcesApi(String accessToken) {
-    return new ResourcesApi(getApiClient(accessToken));
-  }
```

Comment:
If you wanted to, you could've kept these methods and had them forward to the new APIs. Either way works fine of course.
```
  public ResourcesApi samResourcesApi(String accessToken) {
    return samApiService.resourcesApi(accessToken);
  }
```

Reply:
That's true, though I suspect that they only existed to help with the spying in `SamIamTest` so I'll leave them out.